libs-scala/resources/src/test/suite/scala/com/digitalasset/resources/ResourceOwnerSpec.scala

```diff
@@ -734,6 +733,56 @@ final class ResourceOwnerSpec extends AsyncWordSpec with Matchers {
         )
       }
     }
+
+    "convert to a ResourceOwner, which ensures, that no TimerTask is running after the resource is released" in {
+      val finishLongTakingTask = Promise[Unit]()
+      val timerResource = Factories.forTimer(() => new Timer("test timer")).acquire()
+      for {
+        timer <- timerResource.asFuture
+        (timerReleased, extraTimerTaskStarted) = {
+          timer.schedule(
+            new TimerTask {
+              override def run(): Unit = {
+                Await.result(finishLongTakingTask.future, Duration(10, "seconds"))
+              }
+            },
+            0L,
+          )
```

Comment:
nitpick: the whole `timer.schedule` could be extracted into a small `def` and reused 3 more times :) not a big concern though